txpipe/diagnostics.py

```diff
@@ -455,6 +471,84 @@ def plot_snr_histogram(self):
         plt.ylim(0,1.1*max(count1))
         fig.close()
 
+    def plot_response_histograms(self):
+        import matplotlib.pyplot as plt
+        if self.comm:
+            import mpi4py.MPI
+        size = 40
+        # This seems to be a reasonable range, though there are samples
+        # with extremely high values
+        edges = np.linspace(-3, 3, size+1)
+        mid = 0.5*(edges[1:] + edges[:-1])
+        width = edges[1] - edges[0]
+        # count of objects
+        counts = np.zeros((2,2,size))
+        # make a separate histogram of the shear-sample-selected
+        # objects
+        counts_s = np.zeros((2,2,size))
+        while True:
+            data = yield
+
+            if data is None:
+                break
```

Comment:
This might just be me being dumb, but what does this do?

Reply:
These are co-routines - functions that you can pause and come back to later (in this case when we load the next chunk of data.  There's an explanation in the run method, and I'll talk about them at the remote meeting DE school in July!